pages/product/[id].js

```diff
@@ -0,0 +1,30 @@
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import Image from "next/image";
+import Link from "next/link";
+
+export default function ProductDetail() {
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: product, error, isLoading } = useSWR(`/api/shoppinglist/${id}`);
+
+  if (!id || isLoading) return <p>Loading product...</p>;
+  if (error) return <p>Error loading product</p>;
```

Comment:
Swap the error check to before the loading check. Otherwise users may never see the error message.

```suggestion
  if (error) return <p>Error loading product</p>;
  if (!id || isLoading) return <p>Loading product...</p>;
```